website/docs/components/table/partials/guidelines/guidelines.md

```diff
@@ -155,4 +155,45 @@ Row placement determines the visual styling based on where the row is placed rel
 - We recommend using `medium` cell density by default.
 - If content is complex or a smaller data set (e.g., a Table of basic user data), `tall` cell density allows for more breathing room surrounding the content.
 - If content is largely string/text-based, `short` allows for more content to be displayed within the page.
-- While denser content allows for more rows to be displayed within a single page, it also makes comprehension and scanning more difficult.
\ No newline at end of file
+- While denser content allows for more rows to be displayed within a single page, it also makes comprehension and scanning more difficult.
+
+## Multi-selection
+
+A multi-select table includes checkboxes enabling users to select multiple rows in a table for purposes of performing bulk operations. Checking or unchecking the checkbox in the table header either selects or deselects the checkboxes on each row in the table body. Individual checkboxes in the rows can also be selected or deselected.
+
+Multi-selection consists of: 
+
+1. `Header Column / Select`: Used in the table's header row. This acts as the parent checkbox, allowing the selection or deselection of all child rows in a single table simultaneously.
+
+![Example of multi-selection in a table header](/assets/components/table/multi-select-header.png)
+
+2. `Cell / Select`: Used in each table row allowing for the selection of an individual row.
+
+![Example of multi-selection within table cells](/assets/components/table/multi-select-cells.png)
+
+
+### Intended interaction
+
+- When individual rows are selected, the parent checkbox in the Table header changes to display an indeterminate state. 
+- When no or some rows (but not all) are selected on a single Table, clicking the parent checkbox in the Table header will change to display as checked and all rows on that page will be selected.
+- When all rows are selected on a single Table, the parent checkbox in the header appears as checked. Clicking the parent checkbox will deselect all rows on that page.
+- An additional action, outside of the Table, is needed in order to select all rows across all pages.
```

Comment:
@KristinLBradley and @hashicorp/hds-design 
--
In the multi-select consist... area above, I have written out the following: 

1. `Header Column / Select`: Used in the table's header row. This acts as the parent checkbox, allowing the selection or deselection of all child rows in a single table simultaneously.

While here, I have written out the intended interaction, which covers the expected interactions for the parent checkbox. Would adding in this section something along the lines of: 

While we recommend this expected interaction for the parent checkbox, on the application level there is flexibility for what occurs on the end user experience. For example, the parent checkbox can select all child checkboxes in the current view.

Reply:
I'm not sure exactly what this means and not sure if we need to include other info around this: 

"While we recommend this expected interaction for the parent checkbox, on the application level, there is flexibility for what occurs in the end-user experience. For example, the parent checkbox can select all child checkboxes in the Table."

Reply:
I think what you updated that point to works 👍

Regarding the flexibility of the parent checkbox, I'm not sure if this flexibility actually exists within the component. @KristinLBradley is it possible for a consumer to alter the functionality of the parent checkbox to select _all_ rows in a paginated table?

I don't think I would include something like that for now, but maybe there's a better way to cover the flexibility?

Reply:
It's not built in. They would have to write a feature to track the state and use that to update everything and we would need to provide a way to select an initial checkbox state from a parameter that they provide. The current implementation doesn't track state. I could possibly look into this.

Reply:
Yeah, maybe keeping that in mind for when we begin pattern documentation for this would come in handy, but i'll keep the current documentation for this section then. Thank you!

Reply:
@andgen404 we can look into it, if it's achievable with the current implementation (I imagine it would be, since they control the state, but we have to double check)

@KristinLBradley we need to discuss and clarify what we mean with "state" to avoid confusion between us: there are different possible "states" that we can refer to: the initial state (if a row is selected or not), the current state (if a row is selected or not in the DOM), the persisted state (eg if a user refresh the page), etc.
Plus, we have to consider the state of a row single row (that may change atomically), vs the state of the entire table (that can change all at once, via the "select all", or when one of the rows is updated).